src/ImageService.Broker/Consumers/CreateImagesConsumer.cs

```diff
@@ -19,35 +21,43 @@ public class CreateImagesConsumer : IConsumer<ICreateImagesRequest>
     private readonly IImageRepository _imageRepository;
     private readonly IDbImageMapper _dbImageMapper;
     private readonly IImageResizeHelper _resizeHelper;
-    private readonly ILogger<CreateImagesConsumer> _logger;
-
+    private readonly ILogger<CreateImagesConsumer> _logger;
+
     private async Task<object> CreateImagesAsync(ICreateImagesRequest request)
     {
       List<DbImage> dbImages = new();
-      List<Guid> previewIds = new();
-      DbImage dbImage;
-      DbImage dbPrewiewImage;
-      (bool isSuccess, string resizedContent, string extension) resizeResult;
+      List<Guid> previewIds = new(); 
 
       foreach (CreateImageData createImage in request.Images)
       {
-        dbImage = _dbImageMapper.Map(createImage, request.CreatedBy);
-        resizeResult = await _resizeHelper.ResizeAsync(createImage.Content, createImage.Extension);
+        (bool isSuccess, string resizedContent, string extension) imageResizeResult =
+          request.ImageSource.Equals(ImageSource.User) 
+            ? await _resizeHelper.ResizeAsync(createImage.Content, createImage.Extension, (int)ImageSizes.Middle)
+            : await _resizeHelper.ResizeAsync(createImage.Content, createImage.Extension, (int)ImageSizes.Big);
+
+        (bool isSuccess, string resizedContent, string extension) previewResizeResult =
+          request.ImageSource.Equals(ImageSource.User)
+            ? await _resizeHelper.ResizeForPreviewAsync(createImage.Content, createImage.Extension)
+            : await _resizeHelper.ResizeForPreviewAsync(createImage.Content, createImage.Extension, 4, 3);
 
-        if (!resizeResult.isSuccess)
+        if (!imageResizeResult.isSuccess || !previewResizeResult.isSuccess)
         {
           _logger.LogError("Error while resize image.");
           return null;
         }
```

Comment:
maybe move it to 37'th line and do preview resizing after check that image Resize went without errors? or do these resizing in parallel?

Reply:
done